tools/reviewer/web/src/app/dashboard/reviews-panel/review/person-list/person-list.component.ts

```diff
@@ -1,97 +1,101 @@
+  ngOnInit() {
+    switch (this.title) {
+      case 'Reviewers':
+        this.isNeedAttention = true;
         break;
-      case 'cc':
-        this.persons = this.diff.getCcList();
+
+      case 'CC':
+        this.isNeedAttention = false;
         break;
+
       default:
-        throw new Error('Unsupported property');
+        throw new Error('PersonListComponent: unsupported title');
     }
 
-    this.propertyValue = this.persons.join(', ');
+    this.getPropertyValue();
+  }
+
+  // Get property value from the Diff
+  getPropertyValue(): void {
+    this.reviewers = this.diff.getReviewerList()
+      .filter(reviewer => {
+        return reviewer.getNeedsattention() === this.isNeedAttention;
+      });
+
+    this.reviewerEmails = this.reviewers
+      .map(reviewer => reviewer.getEmail())
+      .join(', ');
   }
 
   // Save the new value of property and update Diff
-  savePropertyValue(): void {
-    const persons = this.propertyValue.split(',');
-    // Trim white spaces
-
-    // WARNING: Bad practice
-    // Do not use: object[propertyName]
-    // - It forces to use any.
-    // - It causes undetected errors.
-    // - It's difficult to maintain the code.
-    // use instead: object.field
-    // TODO: refactor this component ('switch' is not the best way too)
-
-    const newPersonsList = persons.map(v => v.trim()).filter(v => v.length);
-    let length;
-
-    switch (this.property) {
-      case 'reviewers':
-        this.diff.setReviewersList(newPersonsList);
-        length = this.diff.getReviewersList().length;
-        break;
-      case 'cc':
-        this.diff.setReviewersList(newPersonsList);
-        length = this.diff.getReviewersList().length;
-        break;
-      default:
-        throw new Error('Unsupported property');
-    }
+  saveAttentionList(): void {
+    let reviewers: Reviewer[] = [];
 
-    if (length === 0) {
-      this.onRemoveProperty.emit(this.property);
-    } else {
-      this.onUpdateDiff.emit(this.diff);
-    }
+    this.reviewerEmails
+      .split(',')
+      .map(reviewer => reviewer.trim())
+      .filter(reviewer => reviewer.length)
+      .forEach(email => {
+        const reviewer = new Reviewer();
+        reviewer.setEmail(email);
+        reviewer.setApproved(false);
+        reviewer.setNeedsattention(this.isNeedAttention);
+        reviewers.push(reviewer);
+      });
+
+    const otherReviewers = this.diff.getReviewerList().filter(reviewer =>
+      reviewer.getNeedsattention() !== this.isNeedAttention
+    );
+
+    reviewers = reviewers.concat(otherReviewers);
+    this.diff.setReviewerList(reviewers);
+
+    this.reviewService.saveLocalDiff(this.diff);
   }
 
-  // Save needAttentionOf list
-  saveAttentionList(name: string): void {
-    this.onAddToAttentionList.emit(name);
+  // Add to CC, if reviewer is inculed in Reviewers.
+  // Add to Reviewers, if reviewer is inculed in CC.
+  changeAttentionOfReviewer(reviewer: Reviewer): void {
```

Comment:
I think this needs to be changed - let me explain the concepts better.


CC is never in the "Attention set".
CC is a passive observer.

Only Author and Reviewers can be in the Attention Set.

Attention Set is pretty much - who's turn is it to do something with this Diff. Sometimes it's the Author, sometimes it's the Reviewer. It can also be both. It can also be only some of the reviewers but not all.